cantools/database/can/signal.py

```diff
@@ -386,15 +386,17 @@ def comment(self, value: Optional[str]) -> None:
         else:
             self.comments = {None: value}
 
-    def choice_string_to_number(self, string: str) -> int:
-        if self.choices is None:
-            raise ValueError(f"Signal {self.name} has no choices.")
-
-        for choice_number, choice_value in self.choices.items():
-            if str(choice_value) == str(string):
-                return choice_number
-
-        raise KeyError(f"Choice {string} not found in Signal {self.name}.")
+    def set_choices(self, choices: Optional[Choices]) -> None:
+        self.choices: Optional[Choices] = choices
+        self._inverse_choices: Optional[Dict[str, int]] = None
+        if choices is not None:
+            # we simply assume that the choices are invertible
+            self._inverse_choices = { str(x[1]): x[0] for x in choices.items() }
+
+    def choice_to_number(self, choice: Union[str, NamedSignalValue]) -> int:
```

Comment:
I hope you'll remember all these API changes when you create the next release 😄

Reply:
yeah. If not that should not be a too big deal as I'm pretty sure that not too many people use this function as it is pretty low level... (the old function name was a bit of a misnomer IMO, because it also accepted `NamedSignalValue` objects.)